enhancements/20220822-storage-network/20220822-storage-network.md

```diff
@@ -0,0 +1,108 @@
+# Storage Network
+
+## Summary
+
+Longhorn storage will need sync data to every replica engine. The sync traffic will be heavy and occupy too many bandwidth from management interface. To keep management interface being reliable. Storage Network will help to separate traffic to another interfaces.
+
+### Related Issues
+
+https://github.com/harvester/harvester/issues/1055
+https://github.com/longhorn/longhorn/issues/2285
+
+## Motivation
+
+### Goals
+
+- Longhorn will sync replica from storage network
+
+### Non-goals [optional]
+
+- Set storage network in the installation stage
+
+## Proposal
+
+### User Stories
+
+Assume Harvester nodes will have 4 network interfaces (eth0, eth1, eth2, eth3).
+Users want to have eth0 and eth1 to be management interface with HA.
+And eth2, eth3 are for VM and Storage Network.
+
+VM and Storage Network will have different VLAN ID.
+
+#### Story 1
+
+VM and Storage Network will use the same interfaces but different VLAN IDs.
+
+- VM and Storage want to share eth2 and eth3.
+- Create VlanConfig with eth2 and eth3
+- Create Network-Attatchment-Definition for VM
+- Create Network-Attatchment-Definition with Whereabouts config for Storage
+
+#### Story 2
+
+VM and Storage Network will use different interfaces.
+
+- VM use eth2 and eth3
+- Storage uses eth4 and eth5
+- Create a new VlanConfig with eth4 and eth5
+- Create Network-Attatchment-Definition with Whereabouts config for Storage only
+
+### User Experience In Detail
+
+#### Manual
+
+1. Users have to shutdown all VMs and stop all Pods attached with Longhorn volumes manually before setting Longhorn Storage Network. (Including Grafana and Prometheus)
```

Comment:
The `Prometheus` and `Grafana` POD are using PVC to store data, and there will be `alertmanager` and `logging`, which also use PVC.

It is a bit complex to stop those PODs, user needs to `scale down` the related deployment/replicaset, meanwhile, the `rancherd/fleet-agent` watches `managedchart` and may revert any downstream change (the `scale down`). That will cause the PODs starts again earlier than expected.

Careful tryout and test is needed.

Reply:
I already tested prometheus and grafana.
but i didn't test other new pods.
If you have new pods with LH volume, please help us to describe how to scale down those pods.

Reply:
You could refer to this controller

https://github.com/harvester/harvester/pull/2706/files#diff-87504a1be112ef39a8cdaf8814e6bc2da202ca091364097a26d589d62f8c9248